contracts/core/OptionsPremiumPricer.sol

```diff
@@ -0,0 +1,139 @@
+//SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.7.3;
+
+import {IVolatilityOracle} from "../interfaces/IVolatilityOracle.sol";
+import {IPriceOracle} from "../interfaces/IPriceOracle.sol";
+import {ICompToken} from "../interfaces/ICompToken.sol";
+import {IERC20Detailed} from "../interfaces/IERC20Detailed.sol";
+import {Math} from "../libraries/Math.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {Welford} from "../libraries/Welford.sol";
+
+contract OptionsPremiumPricer {
+    using SafeMath for uint256;
+
+    /**
+     * Immutables
+     */
+    IVolatilityOracle public immutable volatilityOracle;
+    IPriceOracle public immutable priceOracle;
+    ICompToken public immutable cToken;
+
+    constructor(
+        address _volatilityOracle,
+        address _priceOracle,
+        address _cToken
+    ) {
+        require(_volatilityOracle != address(0), "!_volatilityOracle");
+        require(_priceOracle != address(0), "!_priceOracle");
+        require(_cToken != address(0), "!_cToken");
+        volatilityOracle = IVolatilityOracle(_volatilityOracle);
+        priceOracle = IPriceOracle(_priceOracle);
+        cToken = ICompToken(_cToken);
+    }
+
+    /**
+     * @notice Calculates the premium of the provided option using Black-Scholes
+     * References for Black-Scholes:
+       https://www.macroption.com/black-scholes-formula/
+       https://www.investopedia.com/terms/b/blackscholes.asp
+     * @param st is the strike price of the option
+     * @param t is the time to expiration in years
+     * @param isPut is whether the option is a put option
+     */
+    function getPremium(
+        uint256 st,
+        uint256 t,
+        bool isPut
+    ) external returns (uint256 premium) {
+        uint256 sp = priceOracle.latestAnswer().div(priceOracle.decimals());
+        uint256 v = volatilityOracle.vol();
```

Comment:
We need to check if the the calculations use the volatility per period (`volatilityOracle.vol`) or annualized volatility (`volatilityOracle.annualizedVol`)

Reply:
in [wikipedia](https://en.wikipedia.org/wiki/Black%E2%80%93Scholes_model) for black scholes there is no mention that volatility is annualized. It explicitly states that risk free rate and drift rate is annualized, so dont think volatility is annualized. Also, volatility in this case is the _standard deviation of log returns_. VolOracle do esn't do log returns right?

Reply:
I'm actually not sure if the volatility input passed should be annualized, weekly or over 6 hours (what `vol` returns now). Will need to test this out.

>VolOracle do esn't do log returns right?

Yea the VolOracle just takes the most basic definition of volatility which is the standard deviation of returns. There are a few conflicting definitions out there that involve using the natural log of returns

Reply:
do you need anything from me here?

Reply:
Nope, we can proceed with what we have here first